Domain/Messages/ReplenishHandCardMessage.cs

```diff
@@ -0,0 +1,30 @@
+﻿using Domain.PlayingCards;
+using System.Collections.Generic;
+
+namespace Domain.Messages
+{
+    public class ReplenishHandCardMessage : TakeCardsMessage
+    {
+        public BangGameCard ReplenishHandCard { get; }
+
+        public ReplenishHandCardMessage(BangGameCard replenishHand) : base()
+        {
+            if (replenishHand.Type is StagecoachCardType)
```

Comment:
I don't like idea to use _if_ in constructor (except argument validation). 
I believe it will be better to introduce two classes that inherit ReplenishHandCardMessage

Reply:
done